src/result_processor.c

```diff
@@ -1029,30 +1060,32 @@ int rpbufferNext_ValidateAndYield(ResultProcessor *rp, SearchResult *result_outp
 
   }
 
-  return RS_RESULT_EOF;
+  return RPBufferAndLocker_ResetAndReturnLastCode(rp);
```

Comment:
Handle when buffer rp is EOF but upstream is not (need to get more valid results)